app/src/main/java/org/groundplatform/android/ui/map/gms/features/FeatureManager.kt

```diff
@@ -136,4 +137,33 @@ constructor(
   fun onCameraIdle() {
     clusterManager.onCameraIdle()
   }
+
+  /**
+   * Generic in-place polyline updater. Callers that do incremental updates (e.g. while dragging)
+   * can use this without forcing a full add/remove diff.
+   */
+  fun updateLineString(
+    tag: Feature.Tag,
+    geometry: LineString,
+    style: Feature.Style,
+    selected: Boolean,
+    tooltipText: String?,
+  ) {
+    coroutineScope.launch {
+      mapsItemManager.updateLineString(tag, geometry, style, selected, tooltipText)
+      // keep indices consistent for future diffs
+      featuresByTag[tag]?.let { prev ->
+        val updated =
+          prev.copy(
+            geometry = geometry,
+            style = style,
+            selected = selected,
+            tooltipText = tooltipText ?: prev.tooltipText,
+          )
```

Comment:
What's the point of this? I thought we already reused the tag from previous feature. So, it seems redundant to me